### PR TITLE
Update competencies table

### PR DESCRIPTION
**Pull request by denizedaza**
## Proposed changes

Closes issue #370 and is related to #369. This adds the foreign key of `categories_id` from the newly created `categories` table into the existing `competencies` table. This builds on and incorporates the latest ES Lint checks and the migration from #369

## Screenshots
Migration build on flyway:
image

Updated database:
image
image

## Checklist

- [x] Are the issues being addressed linked to this PR?
- [x] Do all commit messages start with the issue number?
- [x] Are all code changes sufficiently tested?
- [x] Are there screenshots for UI changes?

### CODE REVIEW

**Review by daveytech — APPROVED** [review]
This satisfies the issue. Thank you for including the screenshots.

File: db/migration/V016__add_category_id_to_competencies.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE competencies
+  ADD COLUMN category_id BIGINT,
```
Comment (sheacronin) [inline db/migration/V016__add_category_id_to_competencies.sql:2]: Do we want to add an `INDEX` here as well? I noticed that other tables that have foreign keys have done this (such as in the V_011 and V_006 migration files). I believe it is supposed to increase the speed of querying the table, but I'm not sure if it is necessary or not.

Comment (daveytech) [reply db/migration/V016__add_category_id_to_competencies.sql:2]: Good Call @sheacronin ! @denizedaza Let's add it in to be consistent before merge.

Comment (denizedaza) [reply db/migration/V016__add_category_id_to_competencies.sql:2]: On it!

Comment (denizedaza) [reply db/migration/V016__add_category_id_to_competencies.sql:2]: If I'm understanding the `INDEX` property correctly, would it be something like this?
image

Comment (daveytech) [reply db/migration/V016__add_category_id_to_competencies.sql:2]: Yes that fits the bill!

**Review by sheacronin — APPROVED** [review]
This code appears to add the `category_id` foreign key to the competencies table as intended, and the screenshots show the correctly updated database

**Pull request merged**